components/match2/commons/match_group_input.lua

```diff
@@ -487,6 +491,63 @@ function MatchGroupInput.fetchStandaloneMatch(matchId)
 	end)
 end
 
+---@class readOpponentOptions: MatchGroupInputReadPlayersOfTeamOptions
+---@field pagifyOpponentName boolean?
+---@field pagifyPlayerNames boolean?
+---@field syncPlayer boolean?
+
+---@param match table
+---@param opponentIndex integer
+---@param options readOpponentOptions
+---@return table?
+function MatchGroupInput.readOpponent(match, opponentIndex, options)
+	options = options or {}
+	local opponentInput = Json.parseIfString(Table.extract(match, 'opponent' .. opponentIndex))
+	-- let the custom handle empty input, some might fill it with blank opponents, some might nto fill it
+	if not opponentInput then return end
+
+	local opponent = Opponent.readOpponentArgs(opponentInput)
+	if Opponent.isBye(opponent) then
+		return {type = Opponent.literal, name = 'BYE'}
+	end
+
+	---@type number|string
+	local resolveDate = match.timestamp
+	-- If date is default date, resolve using tournament dates instead
+	-- default date indicates that the match is missing a date
+	-- In order to get correct child team template, we will use an approximately date and not the default date
+	if resolveDate == DateExt.defaultTimestamp then
+		resolveDate = Variables.varDefaultMulti('tournament_enddate', 'tournament_startdate', NOW)
```

Comment:
I think NOW can be dropped in favor of nil value, as TT-ext should use NOW when there's no date provided

Reply:
need to check on the player sync too

Reply:
```suggestion
		resolveDate = DateExt.getContextualDate()
```